src/tests/tracing/eventpipe/common/EventPipeProvider.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+
+namespace Tracing.Tests.Common
+{
+    public sealed class EventPipeProvider
+    {
+        public long Keywords { get; }
+        public EventLevel EventLevel { get; }
+        public string Name { get; }
+        public IDictionary<string, string> Arguments { get; }
+
+        public EventPipeProvider(string name, EventLevel eventLevel, long keywords = 0xF00000000000, IDictionary<string, string> arguments = null)
+        {
+            Name = name;
+            EventLevel = eventLevel;
+            Keywords = keywords;
+            Arguments = arguments;
+        }
+
+        internal string GetArgumentString()
```

Comment:
If the contents of the messages sent don't actually matter for these tests, then perhaps this can be removed as well